### ISSUE TITLE
How to dynamically change eaf-open-camera save path?

### ISSUE THREAD

**Issue by stardiviner** [manateelazycat/emacs-application-framework#206]
I try to change it by following advice:

```emacs-lisp
(defun eaf-camera-change-save-path ()
  "Change `eaf-open-camera' save path."
  (eaf-setq eaf-camera-save-path default-directory))

(advice-add 'eaf-open-camera :before #'eaf-camera-change-save-path)
```
But it seems does not work. I don't know how to change this path.

**Comment by MatthewZMD** [comment]
Why do you need advice? Simply put (eaf-setq eaf-camera-save-path default-directory) to your init.el

**Comment by stardiviner** [comment]
I need make it dynamic, I want to add it to org-attach dispatcher, so I can save it with org-attach and add org-stored-links.

**Comment by MatthewZMD** [comment]
I just had a quick test, when you `advice-add` prior to starting any EAF process (i.e there is no `*eaf*` buffer), then `advice-add` works with no problem - it changes `eaf-camera-save-path` when calling `eaf-open-camera`.

Ah, I believe the cause is commits 3dad99f and 27976bd. These commits altered the design of how `eaf-var-list` variables are being passed to the Python side, to improve efficiency and fix a previous bug. The biproduct of such design is that Python side only receives these variables at the start of process creation, which means when you have `*eaf*` process running, any futher `eaf-setq` does not actually notify the Python side. You will have to restart all EAF processes for it to take effect again.

In other words, it is not as dynamic as users expect.

I will see what can I do to fix this issue.

**Comment by stardiviner** [comment]
I see, thanks @MatthewZMD

**Comment by manateelazycat** [comment]
This issue has fixed in newest version
